cli/2024-04-azure-workload-identity.md

```diff
@@ -0,0 +1,264 @@
+
+Radius supports Azure service principal identity to deploy and manage Azure resources. However, this adds an extra task for operators to rotate the credentials for the identity. A workload identity or federated identity helps us avoid this maintenance challenge of manually managing the credentials and eliminates the risk of exposing secrets or having certificates expire. 
+
+The goal of the scenario is to enable infrastructure operators to configure Azure workload identity support for Radius to deploy and manage Azure resources.
+
+## Terms and definitions
+
+1. **Entra ID Application** is an application that is registered in Azure Active Directory (Azure AD) and is used to authenticate with Azure resources.
+
+1. **Azure Service Principal** is the local representation of an Entra ID Application object in a tenant or directory. It's the identity of the application instance. Service principals define application access and resources the application accesses.
+
+1. **Azure Workload Identity** is an alternative form of Azure authentication  that removes the need to manage secrets. Azure Workload Identity uses the same service principal or managed identity to trust tokens from an external identity provider (IdP). Once that trust relationship is created, your external software workload exchanges trusted tokens from the external IdP for access tokens from the Microsoft Identity platform. Your software workload uses that access token to access the protected Azure resources to which the workload has been granted access. This eliminates the burden of managing and rotating the secrets.
+
+1. **Azure AD Workload Identity mutating admission webhook (AzWI mutating admission webhook)** is a pod on the Kubernetes cluster that intercepts the pod creation request and injects the signed service account token into the pod. This signed service account token is used by the Azure provider to authenticate with Azure resources.
+
+
+## Objectives
+
+> **Issue Reference: https://github.com/radius-project/radius/issues/7308**
+
+### Goals
+
+* Radius users can configure Azure provider to use Azure workload identity for authentication.
+* Azure workload identity can be configured via interactive experience.
+* Azure workload identity can be configured manually (scriptable).
+* Radius users can deploy and manage Azure resources without using a Entra ID Application client secret.
+
+### Non-goals
+
+* **Azure Managed Identity support**
+    - Managed Identity support for AKS is a much older feature, and Workload Identity is the replacement. We're not planning to support Managed Identity, as any user should be able to upgrade to Workload Identity.
+
+* **AWS Workload Identity (IRSA) support**
+    - Will be tracked as separate work: https://github.com/radius-project/radius/issues/7618
+
```

Comment:
I really like how the non-goals say *why* these are non-goals. This is a practice we should keep following for the improved clarity.